### DISCUSSION TITLE
Adding `asin` / "configuration is invalid" error

### DISCUSSION THREAD

**Discussion by benlabbeus** [MilesCranmer/PySR#345] [Q&A] [answered]
Hi sorry for the repeat questions, a bit new to this.

I'm trying to add asin as a unary operator, I have the following:

```
unary_operators=["cos", "sin", "asin(x)=asin(x)", "tan", "atan", "abs", "sqrt"],
extra_sympy_mappings={"atan2": lambda x, y: sympy.atan(x,y),
                          "asin": lambda x: sympy.asin(x)},
```

And I'm getting the error

`Your configuration is invalid - one of your operators (asin) is not well-defined over the real line. You can get around this by returning `NaN` for invalid inputs.`

Should I be defining the unary operator like this?

`unary_operators=["cos", "sin", "asin(x)=x", "tan", "atan", "abs", "sqrt"],`

**Comment by MilesCranmer** [comment, marked as answer]
You are almost there. Since `asin` is not valid for `abs(x)>1`, you need to create a new operator that either (1) clips the input to be valid, or (2) returns a NaN for invalid input.

I generally prefer (2) as the output equations won't have weird steps in them. You could do this:

```python
"my_asin(x) = abs(x) > 1 ? convert(typeof(x), NaN) : asin(x)"
```

and then just give it the normal `sympy` definition by for `"my_asin"` instead.

**Reply by benlabbeus** [reply]
Awesome!! We're rolling! Thank you!!
